src/mito2/src/sst/index/bloom_filter/creator.rs

```diff
@@ -189,14 +189,20 @@ impl BloomFilterIndexer {
         self.do_cleanup().await
     }
 
-    async fn do_update(&mut self, batch: &Batch) -> Result<()> {
+    async fn do_update(&mut self, batch: &mut Batch) -> Result<()> {
         let mut guard = self.stats.record_update();
 
         let n = batch.num_rows();
         guard.inc_row_count(n);
 
         // TODO(weny, zhenchi): lazy decode
-        let values = self.codec.decode(batch.primary_key())?;
+        if batch.pk_values().is_none() {
+            let values = self.codec.decode(batch.primary_key())?;
+            batch.set_pk_values(values);
+        }
```

Comment:
Do we use the decoded values outside of the creator?

Reply:
Nope, but other index creators can reuse the decoded values.